internal/slicer/slicer.go

```diff
@@ -28,8 +28,9 @@ func Run(options *RunOptions) error {
 	archives := make(map[string]archive.Archive)
 	extract := make(map[string]map[string][]deb.ExtractInfo)
 	pathInfos := make(map[string]setup.PathInfo)
-	knownPaths := make(map[string]bool)
+	report := NewReport(options.TargetDir)
```

Comment:
I just realised that we are not actually **yet** writing the report anywhere. So please discard the above comment about tests, for now.

But since we are not writing the report anywhere, should the changes in this file be in this PR or come in a next one?

Reply:
I am including the changes gradually to see how it all fits together. I think it is difficult to review this PR without seeing how the report will be used. In the next PRs we will add the rest of the usage and add tests both for the output of report after slicing and checking the chisel.db file from disk.